torch/_functorch/aot_autograd.py

```diff
@@ -2474,6 +2630,14 @@ def aot_dispatch_autograd(flat_fn, flat_args: List[Any], aot_config: AOTConfig,
     fx_g.graph.eliminate_dead_code()
     fx_g.recompile()
 
+    if aot_config.is_export:
+        # For export, we return the joint directly
+        # TODO: in AOTAutograd, we create metadata like _indices_of_inps_to_detach to detect
+        # when we need to manually detach() some inputs in the forward.
+        # Higher order ops might eventually need to do the same.
+        return fx_g
```

Comment:
slightly concern about returning a different type for `aot_dispatch_autograd`.
The function return signature will be confused by this... 

I wonder if it can be refactored into sth like this
```
def aot_dispatch_autograd_graph_version -> GraphModule:
      ....
      return fx_g

def aot_dispatch_autograd -> Callable:
     fx_g = aot_dispatch_autograd_graph_version(...)
     fn = create_runtime_wrapper(fx_g)
     return fn
```
Please also feel free to ignore this nitpick, coz I know it might be hard to do so in practice.

Reply:
Agreed - I originally planned to push this off and wait for more clarity on how functionalized RNG fits into the export picture. But I may as well do this refactor now (and I'll leave functionalized RNG only into the callable version for now).

Reply:
updated